src/main/java/com/github/manolo8/darkbot/core/manager/GroupManager.java

```diff
@@ -67,6 +69,9 @@ public void update() {
     }
 
     public void tick() {
+        if (group.isValid()) lastValidTime = System.currentTimeMillis();
+        if (!isLoaded()) return;
+
         if (group.address == 0) return;
```

Comment:
```suggestion
        if (group.address == 0) return;

        if (group.isValid()) lastValidTime = System.currentTimeMillis();
        else if (System.currentTimeMillis() - lastValidTime < 10_000L) return; // Wait until reacting to group being invalid
```

from a logic pov, if the address is 0 you don't even need to check for if the group is valid,there is just no group whatsoever, and so, it shouldn't be marked as valid or even try to tick.

Second of all i'd avoid adding an extra isLoaded function for this as it can be solved with just an if-else

Reply:
okay, this one is missing a `&& System.currentTimeMillis != lastValidTime`